manifests/ulimit.pp

```diff
@@ -27,31 +27,14 @@
     }
   }
 
-  file { "/etc/systemd/system/${redis::service_name}.service.d/":
-    ensure                  => 'directory',
-    owner                   => 'root',
-    group                   => 'root',
-    selinux_ignore_defaults => true,
+  systemd::service_limits { "${redis::service_name}.service":
+    limits          => {
+      'LimitNOFILE' => $redis::ulimit,
+    },
+    restart_service => false,
   }
 
   file { "/etc/systemd/system/${redis::service_name}.service.d/limit.conf":
-    ensure => file,
-    owner  => 'root',
-    group  => 'root',
-    mode   => '0444',
-  }
-  augeas { 'Systemd redis ulimit' :
-    incl    => "/etc/systemd/system/${redis::service_name}.service.d/limit.conf",
-    lens    => 'Systemd.lns',
-    changes => [
-      "defnode nofile Service/LimitNOFILE \"\"",
-      "set \$nofile/value \"${redis::ulimit}\"",
-    ],
-  }
-  # Only necessary for Puppet < 6.1.0,
-  # See https://github.com/puppetlabs/puppet/commit/f8d5c60ddb130c6429ff12736bfdb4ae669a9fd4
-  if versioncmp($facts['puppetversion'],'6.1.0') < 0 {
-    include systemd::systemctl::daemon_reload
-    Augeas['Systemd redis ulimit'] ~> Class['systemd::systemctl::daemon_reload']
+    ensure => absent,
   }
```

Comment:
There is no daemon reload code for Puppet < 6.1.0 in `systemd::service_limits`. Interestingly, in combination with `restart_service` that appears to be a bug.

Reply:
The daemon reload is contained in `systemd::dropin_file` which is invoked indirectly by `systemd::service_limits`. As this PR would use the new way of defining the limits regardless of Puppet version, the reload should be handled there.